From 703d48ad41e014edc675f190fb6f8a2e8fe7f032 Mon Sep 17 00:00:00 2001
From: "Juan A. Suarez Romero" <jasuarez@igalia.com>
Date: Thu, 10 May 2018 16:54:21 +0000
Subject: [PATCH] Remove line expression tests

Using #line directive with an expression is considered as undefined
behaviour. Thus, remove these tests.

Removed:
dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_fragment
dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_vertex
dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex

Components: OpenGL

VK-GL-CTS issue: 1145

Change-Id: Ib8cb58a8e712397114eb33d8d36c1957b3f5e9b0
(cherry picked from commit 4ff5a922a15bcdb93e59313221033bee1204be2c)
---
 android/cts/master/gles3-master.txt           |     4 -
 android/cts/master/src/gles2-failures.txt     |     4 -
 android/cts/nyc/gles3-master.txt              |     4 -
 android/cts/nyc/src/gles3-master.txt          |     4 -
 data/gles2/shaders/preprocessor.test          |    28 -
 data/gles3/shaders/preprocessor.test          |    32 -
 .../aosp_mustpass/3.2.3.x/gles2-master.txt    |     4 -
 .../aosp_mustpass/3.2.3.x/gles3-master.txt    |     4 -
 .../3.2.3.x/src/gles2-master.txt              |     4 -
 .../3.2.3.x/src/gles3-master.txt              |     4 -
 .../aosp_mustpass/3.2.4.x/gles2-master.txt    |     4 -
 .../aosp_mustpass/3.2.4.x/gles3-master.txt    |     4 -
 .../aosp_mustpass/3.2.5.x/gles2-master.txt    | 16415 ++++++
 .../aosp_mustpass/3.2.5.x/gles3-master.txt    | 44137 ++++++++++++++++
 .../aosp_mustpass/master/gles2-master.txt     |     4 -
 .../aosp_mustpass/master/gles3-master.txt     |     4 -
 16 files changed, 60552 insertions(+), 108 deletions(-)
 create mode 100644 external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.5.x/gles2-master.txt
 create mode 100644 external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.5.x/gles3-master.txt

diff --git a/android/cts/master/gles3-master.txt b/android/cts/master/gles3-master.txt
index 34f8f3f85974..a1f999ea6dfd 100644
--- a/android/cts/master/gles3-master.txt
+++ b/android/cts/master/gles3-master.txt
@@ -1122,10 +1122,6 @@ dEQP-GLES3.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/android/cts/master/src/gles2-failures.txt b/android/cts/master/src/gles2-failures.txt
index 3d4e7c6fc216..86cc14b79346 100644
--- a/android/cts/master/src/gles2-failures.txt
+++ b/android/cts/master/src/gles2-failures.txt
@@ -1882,16 +1882,12 @@ dEQP-GLES2.functional.shaders.preprocessor.builtin.invalid_line_file_1_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.invalid_line_file_1_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.invalid_line_file_3_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.invalid_line_file_3_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES2.functional.shaders.preprocessor.conditional_inclusion.unary_ops_2_fragment
diff --git a/android/cts/nyc/gles3-master.txt b/android/cts/nyc/gles3-master.txt
index 6b535abf453a..48857d397284 100644
--- a/android/cts/nyc/gles3-master.txt
+++ b/android/cts/nyc/gles3-master.txt
@@ -1122,10 +1122,6 @@ dEQP-GLES3.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/android/cts/nyc/src/gles3-master.txt b/android/cts/nyc/src/gles3-master.txt
index 6b535abf453a..48857d397284 100644
--- a/android/cts/nyc/src/gles3-master.txt
+++ b/android/cts/nyc/src/gles3-master.txt
@@ -1122,10 +1122,6 @@ dEQP-GLES3.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/data/gles2/shaders/preprocessor.test b/data/gles2/shaders/preprocessor.test
index 139e883120d0..f4048cee82b8 100644
--- a/data/gles2/shaders/preprocessor.test
+++ b/data/gles2/shaders/preprocessor.test
@@ -2532,34 +2532,6 @@ group builtin "Built-in Symbol Tests"
 		""
 	end
 
-	case line_expression
-		values { output float out0 = 20.0; }
-		both ""
-			precision mediump float;
-			${DECLARATIONS}
-			void main()
-			{
-			#line +20
-				out0 = float(__LINE__);
-				${OUTPUT}
-			}
-		""
-	end
-
-	case line_and_file_expression
-		values { output vec4 out0 = vec4(243.0, 243.0, 10.0, 10.0); }
-		both ""
-			precision mediump float;
-			${DECLARATIONS}
-			void main()
-			{
-			#line (233 +10) (+10)
-				out0 = vec4(__LINE__, __LINE__, __FILE__, __FILE__);
-				${OUTPUT}
-			}
-		""
-	end
-
 	case line_defined_1
 		values { output float out0 = 4.0; }
 		both ""
diff --git a/data/gles3/shaders/preprocessor.test b/data/gles3/shaders/preprocessor.test
index 1e2d4f1b8054..f290c1ea3554 100644
--- a/data/gles3/shaders/preprocessor.test
+++ b/data/gles3/shaders/preprocessor.test
@@ -3192,38 +3192,6 @@ group builtin "Built-in Symbol Tests"
 		""
 	end
 
-	case line_expression
-		version 300 es
-		values { output float out0 = 20.0; }
-		both ""
-			#version 300 es
-			precision mediump float;
-			${DECLARATIONS}
-			void main()
-			{
-			#line +20
-				out0 = float(__LINE__);
-				${OUTPUT}
-			}
-		""
-	end
-
-	case line_and_file_expression
-		version 300 es
-		values { output vec4 out0 = vec4(243.0, 243.0, 10.0, 10.0); }
-		both ""
-			#version 300 es
-			precision mediump float;
-			${DECLARATIONS}
-			void main()
-			{
-			#line (233 +10) (+10)
-				out0 = vec4(__LINE__, __LINE__, __FILE__, __FILE__);
-				${OUTPUT}
-			}
-		""
-	end
-
 	case line_defined_1
 		version 300 es
 		values { output float out0 = 4.0; }
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles2-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles2-master.txt
index b0082da364e4..0b5b73213cc3 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles2-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles2-master.txt
@@ -437,10 +437,6 @@ dEQP-GLES2.functional.shaders.preprocessor.directive.error_vertex
 dEQP-GLES2.functional.shaders.preprocessor.directive.error_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.empty_line_vertex
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles3-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles3-master.txt
index 17e3649bf66e..6431f5f10d01 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles3-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/gles3-master.txt
@@ -1125,10 +1125,6 @@ dEQP-GLES3.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles2-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles2-master.txt
index 0463eac69ec7..c2ac4f5bc59a 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles2-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles2-master.txt
@@ -449,10 +449,6 @@ dEQP-GLES2.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles3-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles3-master.txt
index 6c905ea4f4ca..1c54bd9dcc93 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles3-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.3.x/src/gles3-master.txt
@@ -1133,10 +1133,6 @@ dEQP-GLES3.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles2-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles2-master.txt
index 182b7ea030c7..d5189b751f6c 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles2-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles2-master.txt
@@ -437,10 +437,6 @@ dEQP-GLES2.functional.shaders.preprocessor.directive.error_vertex
 dEQP-GLES2.functional.shaders.preprocessor.directive.error_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.empty_line_vertex
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles3-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles3-master.txt
index 5a3ca0373af7..0fbed42ab70a 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles3-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/3.2.4.x/gles3-master.txt
@@ -1125,10 +1125,6 @@ dEQP-GLES3.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles2-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles2-master.txt
index b593acc11a5a..36a8a805fd4e 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles2-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles2-master.txt
@@ -449,10 +449,6 @@ dEQP-GLES2.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES2.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES2.functional.shaders.preprocessor.builtin.line_defined_2_vertex
diff --git a/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles3-master.txt b/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles3-master.txt
index e37db5d88532..69860bf60d59 100644
--- a/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles3-master.txt
+++ b/external/openglcts/data/mustpass/gles/aosp_mustpass/master/gles3-master.txt
@@ -1127,10 +1127,6 @@ dEQP-GLES3.functional.shaders.preprocessor.builtin.line_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_expression_fragment
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_vertex
-dEQP-GLES3.functional.shaders.preprocessor.builtin.line_and_file_expression_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_vertex
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_1_fragment
 dEQP-GLES3.functional.shaders.preprocessor.builtin.line_defined_2_vertex
-- 
2.23.0.351.gc4317032e6-goog

